src/reducers/index.js

```diff
@@ -57,6 +57,24 @@ export const removeResource = (state, action) => {
   return validate(newState)
 }
 
+// TODO: could combine setPublishError() and setRetrieveError() with arg for single word diff in error message
+//   for now, shameless green.  (Sandi Metz says hold your nose for 3 occurrences before refactoring)
+export const setPublishError = (state, action) => {
+  const resourceTemplateId = action.payload.resourceTemplateId
+  const reason = action.payload.reason
```

Comment:
Isn't the payload itself the reason? See https://github.com/LD4P/sinopia_editor/pull/1123/files#diff-4e3037116dbc2bc69a473b7ace3d02abR94

Reply:
So the action is this, as your link shows:

```js
export const publishError = reason => ({
  type: 'PUBLISH_ERROR',
  payload: reason,
})
```

But that is some notation foo for 

```
export const publishError = reason => ({
  type: 'PUBLISH_ERROR',
  payload: { reason: reason }
})
```

Which I can show empirically by testing diff references in reducer:

```js
const reason = action.payload
```

results:

![image](https://user-images.githubusercontent.com/96775/62316313-07082800-b44c-11e9-978f-3b54e9691629.png)

while the following passes:

```js
const reason = action.payload?.reason
```